src/main/java/zd/zero/waifu/motivator/plugin/assets/RemoteAssetDefinitionService.kt

```diff
@@ -11,17 +11,44 @@ abstract class RemoteAssetDefinitionService<T : AssetDefinition, U : Asset>(
 ) {
     private val random = Random(System.currentTimeMillis())
 
+    fun getAssetByGroupId(
+        groupId: UUID,
+        category: WaifuAssetCategory
+    ): Optional<U> =
+            remoteAssetManager.supplyLocalAssetDefinitions()
+                .find {
+                    it.groupId == groupId
+                }.toOptional()
+                .map {
+                    resolveAsset(category, it)
+                }.orElseGet {
+                    remoteAssetManager.supplyRemoteAssetDefinitions()
+                        .find { it.groupId == groupId }
+                        .toOptional()
+                        .map { remoteAssetManager.resolveAsset(it) }
+                        .orElseGet {
+                            getRandomAssetByCategory(category)
```

Comment:
Showing something rather than nothing is better in my opinion. 

Also, this covers the case of a motivation asset group missing a category. EG it has a image and sound, but no title.